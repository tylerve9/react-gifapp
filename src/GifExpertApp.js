import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ( {defaultCategories = ['Samurai X']}) => {

    const [categories, setCategories] = useState( defaultCategories);
    //'One Punch', , 'Dragon Ball'


    return (
        <>
            <h2>GifExpertApp</h2>
            <br />
            <AddCategory setCategories={setCategories} />
            <br />
            <hr />
            <br />
            <div>
                { categories.map(cat => <GifGrid key={cat} category={cat} />) }
            </div>
        </>
    );

}

export default GifExpertApp;