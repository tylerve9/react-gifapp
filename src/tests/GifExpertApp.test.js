import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas sobre <GifExpertApp />', () => {

    test('Debe mostrar el componente correctamente <GifExpertApp />  ', () => {
            
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de mostrar una lista de categorias ', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories= {categories} />);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    });
});