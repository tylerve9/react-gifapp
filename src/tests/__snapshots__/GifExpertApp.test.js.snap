// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <br />
  <div>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe mostrar el componente correctamente <GifExpertApp />   1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <br />
  <div>
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
  </div>
</Fragment>
`;
